library/CMakeLists.txt

```diff
@@ -48,11 +48,18 @@ if(HIP_COMPILER STREQUAL "nvcc")
     )
     set(CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
 else()
-    target_link_libraries(rocrand PRIVATE hip::device)
-    if(NOT WIN32)
-        foreach(amdgpu_target ${AMDGPU_TARGETS})
-            target_link_libraries(rocrand PRIVATE --amdgpu-target=${amdgpu_target})
-        endforeach()
+    if(ROCRAND_CMAKE_HIP_LANGUAGE)
+        set_source_files_properties(${rocRAND_SRCS}
+          PROPERTIES
+            LANGUAGE HIP
+        )
+    else()
+        target_link_libraries(rocrand PRIVATE hip::device)
+        if(NOT WIN32)
+            foreach(amdgpu_target ${AMDGPU_TARGETS})
+                target_link_libraries(rocrand PRIVATE --amdgpu-target=${amdgpu_target})
+            endforeach()
+        endif()
```

Comment:
I don't want to change any existing behaviour in this PR.

Reply:
I should add, I think it would be great to simplify the CMake and I fully support removing that code. Just not in this particular PR. I'd like to keep the addition of the new option and changes in the behaviour of existing options separate.